### PR TITLE
Fix unique realm name check to cover default realms

### PR DESCRIPTION
**Pull request by ywangd**
All enabled realms must have unique names. This PR tightened the logic
for ensuring realm name uniqueness. Previously the unique name check
does not cover realms that are automatically enabled.

Relates: #69096

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java [hunk 1]
```diff
@@ -1097,6 +1089,147 @@ public void testWarningsForReservedPrefixedRealmNames() throws Exception {
...
+            )
+        );
+        assertThat(
+            e1.getMessage(),
+            containsString(
+                "Found realm configured with name clashing with the [" + defaultRealm.v2() + "] realm: [" + realmSettingKey + "]"
+            )
+        );
+
+        // It is OK if the explicitly configured realm is disabled
```
Comment (ywangd) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/RealmsTests.java:1126]: This and a few other test scenarios below can be seen as a leniency. Ideally these names should not be used at all. We may want to do something about it, e.g. log warnings. But it is a separate issue from what we are trying to fix here.

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by tvernum — APPROVED** [review]
LGTM

**Review by n1v0lg — APPROVED** [review]
LGTM! Thanks for the fix.

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-2-fips

**Pull request merged**